src/EditorFeatures/CSharpTest/RemoveUnusedParametersAndValues/RemoveUnusedValueAssignmentTests.cs

```diff
@@ -4654,6 +4654,158 @@ void M()
 }", optionName);
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedValues)]
+        public async Task SwitchCase_UnusedValueWithOnlyWrite_PreferDiscard()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    int M(int flag)
+    {
+        switch(flag)
+        {
+            case 0:
+                int [|x|] = M2();
+                return 0;
+
+            default:
+                return flag;
+        }
+    }
+
+    int M2() => 0;
+}",
+@"class C
+{
+    int M(int flag)
+    {
+        switch(flag)
+        {
+            case 0:
+                _ = M2();
+                return 0;
+
+            default:
+                return flag;
+        }
+    }
+
+    int M2() => 0;
+}", options: PreferDiscard);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedValues)]
+        public async Task SwitchCase_UnusedValueWithOnlyWrite_PreferUnusedLocal()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    int M(int flag)
+    {
+        switch(flag)
+        {
+            case 0:
+                int [|x|] = M2();
+                return 0;
+
+            default:
+                return flag;
+        }
+    }
+
+    int M2() => 0;
+}", options: PreferUnusedLocal);
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedValues)]
+        [InlineData(nameof(PreferDiscard))]
+        [InlineData(nameof(PreferUnusedLocal))]
+        public async Task SwitchCase_UnusedConstantValue_WithReadsAndWrites(string optionName)
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    int M(int flag)
+    {
+        switch(flag)
+        {
+            case 0:
+                int [|x|] = 0;
+                x = 1;
+                return x;
+
+            default:
+                return flag;
+        }
+    }
+
+    int M2() => 0;
+}",
+@"class C
+{
+    int M(int flag)
+    {
+        switch(flag)
+        {
+            case 0:
+                int x;
+                x = 1;
```

Comment:
next thing to add to the list: these could be combined :)  (move decl to reference does this i believe).

Reply:
Actually, move decl to reference does not handle this case. I was planning on filing a separate bug for that service.

Reply:
Ok... that's super strange... i could have sworn i made it do that!

When i try i get:

![image](https://user-images.githubusercontent.com/4564579/49261460-30713f80-f3f7-11e8-80aa-bb86cd2555d6.png)

It looks like maybe it works when there is something between the two statements, but not if they're abutting!  Would def be great to improve :)

Reply:
It only works if the declaration is parented by a block syntax, not if the parent is switch section syntax. It bails out [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/MoveDeclarationNearReference/AbstractMoveDeclarationNearReferenceService.State.cs,70)

Reply:
Ah.  Woudl be nice to fix, but obviously not at all critical.